manual/library/builtin.etex

```diff
@@ -190,7 +192,10 @@ exception Failure of string
 \index{Failure@\verb`Failure`}
 \begin{ocamldocdescription}
    Exception raised by library functions to signal that they are
-   undefined on the given arguments.
+   undefined on the given arguments.  The string gives some information
+   to the programmer; it is \emph{bad practice} to pattern match
```

Comment:
I'm personally fine with a stronger wording saying that the string *must not* be pattern-matched. I would not use this place to give a lecture on monadic error handling (and speaking of `result` while we have no stdlib support for it is possibly not helping), but if you write something you like about it I would still merge it.